### ISSUE TITLE
Incorrect mismatched-arg-count for generic types "duplicated" functions

### ISSUE THREAD

**Issue by pedromfedricci** [rust-lang/rust-analyzer#9392]
A minimal reproduction I could come up with, where the incorrect message shows:
```rust
use std::marker::PhantomData;

struct MyStruct<A, B> {
    _pd: PhantomData<*const (A, B)>,
}

trait TraitA {}

impl<A, B> MyStruct<A, B>
where
    A: TraitA,
{
    fn call(&self, _arg: ()) {}
}

impl<B> MyStruct<(), B> {
    fn call(&self) {}
}

fn main() {
    let my_struct: MyStruct<(), ()> = MyStruct { _pd: PhantomData };
    my_struct.call();
}
```
In main's ```my_struct.call()```, rust-analyzer complains that: ```expected 1 argument, found 0```.
My understanding is that it is only looking at the first ```call``` implementation, but ```my_struct``` concrete type actually matches the second ```impl``` block.
Cargo check and build work fine with the sample.
Not sure how this relates to #8479, if at all, I found it to be considerably different to open a new issue.

rust-analyzer version: 3898387f3 2021-06-21 stable
ty.

**Comment by pedromfedricci** [comment]
similar to #7856

**Comment by pedromfedricci** [comment]
Fixed by #13074, released at [2022-08-22](https://github.com/rust-lang/rust-analyzer/releases/2022-08-22).
Awesome!
ty @lowr
